instagram/src/Components/CommentSection/CommentSection.js

```diff
@@ -1,19 +1,70 @@
 import React from 'react';
 import Comment from './Comment';
 import './Comment.css';
+import PropTypes from 'prop-types';
+import CommentInput from "./CommentInput";
+
 class CommentSection extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      comments: props.comments
+      comments: props.comments,
+      comment: " "
     };
   }
-  render() {
-    return (
-      <div className= 'comment-sec'>
-        {this.state.comments.map( (c, i) => <Comment key={i} user={c.username} text={c.text} /> )}
-      </div>
-    );
+  componentDidMount() {
+    const id = this.props.postId;
+    if (localStorage.getItem(id)) {
+      // if still in memory
+      this.setState({
+        comments: JSON.parse(localStorage.getItem(this.props.postId)) // update with memory vals
+      });
+    } else {
+      this.setComments(); // if not in memory, set comments to what is in state in local memory
+    }
+  }
+  componenetWillUnmount() {
+    this.setComments();
   }
+
+  setComments = () => {
+    localStorage.setItem(
```

Comment:
It's great that you're using `localStorage` the only problem I see is upon refresh the comments for one post get passed to all the other posts.